pkg/generator/ingress_translator.go

```diff
@@ -151,7 +153,7 @@ func (translator *IngressTranslator) translateIngress(ctx context.Context, ingre
 						externalPort = port.Port
 						targetPort = port.TargetPort.IntVal
 					}
-					if port.Name == "http2" || port.Name == "h2c" {
+					if !strings.EqualFold(config.GetDisableHTTP2(ingress.Annotations), "true") && (port.Name == "http2" || port.Name == "h2c") {
```

Comment:
Could you move the annotation part to L#160 like? 

```go
                                // Disable HTTP2 if the annotation is specified.
                                if strings.EqualFold(pkgconfig.GetDisableHTTP2(ingress.Annotations), "true") {
                                        http2 = false
                                }
```

It will be much easier to read as "we override the configuration when annotation exists".